src/maconn.ts

```diff
@@ -22,6 +23,11 @@ interface WebRTCMultiaddrConnectionInit {
    * Holds the relevant events timestamps of the connection
    */
   timeline: MultiaddrConnectionTimeline
+
+  /**
+   * Optional metrics counter group for this connection
+   */
+  metrics: CounterGroup | null
```

Comment:
I don't think we should accept `null` as a value, better to mark this optional

```suggestion
  metrics?: CounterGroup
```